File: src/main/java/dev/paigewatson/layoutmaster/data/RollingStockDAL.java
package dev.paigewatson.layoutmaster.data;

import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock;

import java.util.List;

public interface RollingStockDAL
{
    List<RollingStock> getAllRollingStock();

    void deleteAll();

    void delete(RollingStock rollingStockToDelete);

    List<RollingStock> getAllOfAARDesignation(AARDesignation aarDesignation);

    List<RollingStock> getAllOfRoadName(String roadName);

    RollingStock insertRollingStock(RollingStock rollingStockToSave);
}

File: src/main/kotlin/dev/paigewatson/layoutmaster/data/RollingStockDAL.kt
package dev.paigewatson.layoutmaster.data

import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock

interface RollingStockDAL {
    val allRollingStock: List<RollingStock>
    fun deleteAll()
    fun delete(rollingStockToDelete: RollingStock)
    fun getAllOfAARDesignation(aarDesignation: AARDesignation): List<RollingStock>
    fun getAllOfRoadName(roadName: String): List<RollingStock>
    fun insertRollingStock(rollingStockToSave: RollingStock): RollingStock
}
